File: SDKs/amazonaws.com/apigatewayv2/2018-11-29/go/pkg/models/operations/getvpclink.go
package operations

import (
"openapi/pkg/models/shared")

type GetVpcLinkPathParams struct {
    VpcLinkID string `pathParam:"style=simple,explode=false,name=vpcLinkId"`
    
}

type GetVpcLinkHeaders struct {
    XAmzAlgorithm *string `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
    XAmzContentSha256 *string `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
    XAmzCredential *string `header:"style=simple,explode=false,name=X-Amz-Credential"`
    XAmzDate *string `header:"style=simple,explode=false,name=X-Amz-Date"`
    XAmzSecurityToken *string `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
    XAmzSignature *string `header:"style=simple,explode=false,name=X-Amz-Signature"`
    XAmzSignedHeaders *string `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
    
}

type GetVpcLinkRequest struct {
    PathParams GetVpcLinkPathParams 
    Headers GetVpcLinkHeaders 
    
}

type GetVpcLinkResponse struct {
    ContentType string 
    GetVpcLinkResponse *shared.GetVpcLinkResponse 
    NotFoundException *interface{} 
    StatusCode int64 
    TooManyRequestsException *interface{} 
    
}

File: SDKs/amazonaws.com/apigatewayv2/2018-11-29/typescript/src/sdk/models/operations/getvpclink.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class GetVpcLinkPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=vpcLinkId" })
  vpcLinkId: string;
}


export class GetVpcLinkHeaders extends SpeakeasyBase {
  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @Metadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;
}


export class GetVpcLinkRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: GetVpcLinkPathParams;

  @Metadata()
  headers: GetVpcLinkHeaders;
}


export class GetVpcLinkResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  getVpcLinkResponse?: shared.GetVpcLinkResponse;

  @Metadata()
  notFoundException?: any;

  @Metadata()
  statusCode: number;

  @Metadata()
  tooManyRequestsException?: any;
}
